### PR TITLE
GH-1650: As a developer I want to remove the dependency from mangelhaft-cli to n4js-cli

### PR DESCRIPTION
**Pull request by mmews-n4**
closes: #1650

### CODE REVIEW

File: n4js-libs/packages/n4js-mangelhaft-cli/tests/npm-test.sh [hunk 1]
```diff
@@ -12,8 +12,12 @@
 set -e
 cd `dirname $0`/TestPrj
 
+echo "Create test catalog"
+../../../n4js-cli/bin/n4jsc.js compile ../../../.. --testCatalog ./test-catalog.json
```
Comment (szarnekow) [inline n4js-libs/packages/n4js-mangelhaft-cli/tests/npm-test.sh]: Should we have a way to pass the path to n4jsc.js as an env variable?

File: plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/concurrent/XRequestManager.java [hunk 2]
```diff
@@ -22,6 +22,10 @@
 import com.google.inject.Singleton;
 
 /**
+ * Handles requests from the LSP client.
```
Comment (szarnekow) [inline plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/concurrent/XRequestManager.java]: I don't see the synchronization here apart from the comment

Comment (mmews-n4) [reply plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/concurrent/XRequestManager.java]: The synchronization was done already before in commit 8e74fc218c6da26e56288fe271edce2aed063478

File: n4js-libs/packages/n4js-mangelhaft-cli/tests/npm-test.sh (hunk 1 above)
Comment (mmews-n4) [reply n4js-libs/packages/n4js-mangelhaft-cli/tests/npm-test.sh]: I agree and actually it is already available as `n4jsc` (like `mangelhaft-cli`), but I wanted to do it similar to the following call to mangelhaft.

**Pull request merged**